pallets/rmrk-core/src/functions.rs

```diff
@@ -55,29 +57,30 @@ where
 	}
 }
 
-impl<T: Config> Resource<StringLimitOf<T>, T::AccountId> for Pallet<T>
+impl<T: Config> Resource<BoundedVec<u8, T::StringLimit>, T::AccountId, BoundedResource<T::ResourceSymbolLimit>> for Pallet<T>
 where
 	T: pallet_uniques::Config<ClassId = CollectionId, InstanceId = NftId>,
 {
 	fn resource_add(
 		sender: T::AccountId,
 		collection_id: CollectionId,
 		nft_id: NftId,
-		base: Option<BoundedVec<u8, T::StringLimit>>,
+		resource_id: BoundedResource<T::ResourceSymbolLimit>,
+		base: Option<BaseId>,
 		src: Option<BoundedVec<u8, T::StringLimit>>,
 		metadata: Option<BoundedVec<u8, T::StringLimit>>,
-		slot: Option<BoundedVec<u8, T::StringLimit>>,
+		slot: Option<SlotId>,
 		license: Option<BoundedVec<u8, T::StringLimit>>,
 		thumb: Option<BoundedVec<u8, T::StringLimit>>,
-	) -> Result<ResourceId, DispatchError> {
+		parts: Option<Vec<PartId>>,
+	) -> Result<BoundedResource<T::ResourceSymbolLimit>, DispatchError> {
 		let (root_owner, _) = Pallet::<T>::lookup_root_owner(collection_id, nft_id)?;
 
-		let resource_id = Self::get_next_resource_id()?;
-		ensure!(
-			Resources::<T>::get((collection_id, nft_id, resource_id)).is_none(),
-			Error::<T>::ResourceAlreadyExists
-		);
-		// TODO: Check NFT lock status
+		// let resource_id = Self::get_next_resource_id()?;
```

Comment:
Is this no longer needed or a placeholder?

Reply:
no longer needed.  ResourceId is now symbol-based, not incremental

Reply:
fixed in f417d314f752af432a21ab6cfe7300f361c6c63e